packages/discovery-react-components/src/components/TooltipHighlight/TooltipHighlight.tsx

```diff
@@ -0,0 +1,164 @@
+    isOpen: false
+  });
+
+  useEffect(() => {
+    // Compute parameters to overlay the tooltip over the active item
+    const highlightDivRect = parentDiv.current?.getBoundingClientRect() || new DOMRect();
+    const isOpen = tooltipAction.tooltipEvent !== TooltipEvent.LEAVE;
+    const clickRect = tooltipAction.rectActiveElement || new DOMRect();
+    const tooltipRect = new DOMRect(
+      clickRect.x - highlightDivRect.x,
+      clickRect.y - highlightDivRect.y,
+      clickRect?.width,
+      clickRect?.height
+    );
+    const tooltipUpdate = {
+      rectTooltipArea: tooltipRect,
+      tooltipContent: tooltipAction.tooltipContent || <div></div>,
+      isOpen: !!tooltipAction.tooltipContent && isOpen
+    };
+    setTooltipInfo(tooltipUpdate);
+  }, [tooltipAction, setTooltipInfo, parentDiv]);
+
+  return (
+    // Outter div is required to provide tooltip element with position information
+    // "pointerEvents" = "none" so that underlying elements can react to mouse events
+    <div
+      className={cx(baseTooltipPlaceContent)}
+      style={{
+        top: tooltipInfo.rectTooltipArea.y,
+        left: tooltipInfo.rectTooltipArea.x
+      }}
+    >
+      <Tooltip
+        autoOrientation={true}
+        tabIndex={0}
+        showIcon={false}
+        open={tooltipInfo.isOpen}
+        triggerText={
+          <div
+            style={{
+              // Provide size so that tooltip knows boundry where to draw the tooltip
+              width: tooltipInfo.rectTooltipArea.width,
+              height: tooltipInfo.rectTooltipArea.height,
+              pointerEvents: 'none'
+            }}
+          />
+        }
+        children={<div>{tooltipInfo.tooltipContent}</div>}
+      />
+    </div>
+  );
+};
+
+export function calcToolTipContent(
+  facetInfoMap: FacetInfoMap,
+  facetId: string,
+  enrichValue: string
+) {
+  const tableContent = [];
+  let enrichColor = '';
+  let enrichFacetDisplayname = '';
+  if (facetInfoMap[facetId]) {
+    enrichColor = facetInfoMap[facetId].color;
+    enrichFacetDisplayname = facetInfoMap[facetId].displayName;
+    // Will have multiple entries after overlapping is implemented
+    tableContent.push({
+      enrichColor: enrichColor,
+      enrichFacetDisplayname: ellipsisMiddle(enrichFacetDisplayname),
+      enrichValue: ellipsisMiddle(enrichValue)
+    });
+  }
+
+  let tooltipContent = undefined;
+
+  if (enrichFacetDisplayname || enrichValue) {
+    tooltipContent = (
+      <div className={cx(baseTooltipCustomContent)} data-testid="tooltip_highlight_content">
+        <div className={cx(baseTooltipContentHeader)}>
+          {defaultMessages.enrichmentsHeaderLabel} ({tableContent.length})
+        </div>
+        <table>
+          {tableContent.map((oneRow, index) => {
+            let rowBorderClass = {};
+            if (index < tableContent.length - 1) {
+              rowBorderClass = {
+                borderBottom: `1px solid #7A7979`
+              };
```

Comment:
Instead of doing this here, declare the style rules in CSS with a specific classname. Then, the dynamic aspect here just becomes whether to add that classname or not.